### ISSUE TITLE
block-scoped-var CatchClause handling broken

### ISSUE THREAD

**Issue by michaelficarra** [eslint/eslint#922]
``` js
function f(){
  try{}catch(e){}
  e
}
```

Should fail but does not.
